src/components/Utils/filters.js

```diff
@@ -31,3 +31,83 @@ export const mergeFilters = (userFilter, adminAppliedPreFilter) => {
 
   return filterAB;
 };
+
+function isFilterOptionToBeHidden(option, filtersApplied, fieldName) {
+  if (option.count > 0) {
+    return false;
+  }
+
+  if (typeof filtersApplied !== 'undefined'
+      && filtersApplied[fieldName]
+      && filtersApplied[fieldName].selectedValues.includes(option.key)) {
+    return false;
+  }
+
+  return true;
+}
+
+/**
+   * This function updates the counts in the initial set of tab options
+   * calculated from unfiltered data.
+   * It is used to retain field options in the rendering if
+   * they are still checked but their counts are zero.
+   */
+export const updateCountsInInitialTabsOptions = (
+  initialTabsOptions, processedTabsOptions, filtersApplied,
+) => {
+  const updatedTabsOptions = JSON.parse(JSON.stringify(initialTabsOptions));
+  const initialFields = Object.keys(initialTabsOptions);
+  for (let i = 0; i < initialFields.length; i += 1) {
+    const fieldName = initialFields[i];
+    const initialFieldOptions = initialTabsOptions[fieldName].histogram.map(x => x.key);
+    let processedFieldOptions = [];
+    if (Object.prototype.hasOwnProperty.call(processedTabsOptions, fieldName)) {
+      processedFieldOptions = processedTabsOptions[fieldName].histogram.map(x => x.key);
+    }
+
```

Comment:
So I think: for each field in `initialFields`, if there's at least one value selected in `filtersApplied` then just update the counts but keep all the options inside this field.